tests/components/snooz/test_fan.py

```diff
@@ -0,0 +1,268 @@
+):
+    """Test state update events from snooz device."""
+    mock_connected_snooz.device.trigger_state(SnoozDeviceState(False, 64))
+
+    state = hass.states.get(snooz_fan_entity_id)
+    assert ATTR_ASSUMED_STATE not in state.attributes
+    assert state.state == STATE_OFF
+    assert state.attributes[fan.ATTR_PERCENTAGE] == 64
+
+    mock_connected_snooz.device.trigger_state(SnoozDeviceState(True, 12))
+
+    state = hass.states.get(snooz_fan_entity_id)
+    assert ATTR_ASSUMED_STATE not in state.attributes
+    assert state.state == STATE_ON
+    assert state.attributes[fan.ATTR_PERCENTAGE] == 12
+
+    mock_connected_snooz.device.trigger_disconnect()
+
+    state = hass.states.get(snooz_fan_entity_id)
+    assert state.attributes[ATTR_ASSUMED_STATE] is True
+
+
+@pytest.mark.parametrize(
+    "fan_state,percentage",
+    [
+        (STATE_OFF, 1),
+        (STATE_OFF, 22),
+        (STATE_OFF, 50),
+        (STATE_OFF, 99),
+        (STATE_OFF, 100),
+        (STATE_ON, 1),
+        (STATE_ON, 22),
+        (STATE_ON, 50),
+        (STATE_ON, 99),
+        (STATE_ON, 100),
+    ],
+)
+async def test_restore_state(hass: HomeAssistant, fan_state: str, percentage: int):
+    """Tests restoring entity state."""
+    disconnected_device = await create_mock_snooz(
+        connected=False, initial_state=UnknownSnoozState
+    )
+    restored_state = State(
+        "fan.test_restored", fan_state, {fan.ATTR_PERCENTAGE: percentage}
+    )
+
+    with patch(
+        "homeassistant.helpers.restore_state.RestoreEntity.async_get_last_state",
+        return_value=restored_state,
+    ):
+        await create_mock_snooz_config_entry(hass, disconnected_device)
+
+        entity_id = get_fan_entity_id(hass, disconnected_device)
+
+        state = hass.states.get(entity_id)
+        assert state.state == fan_state
+        assert state.attributes[fan.ATTR_PERCENTAGE] == percentage
+        assert state.attributes[ATTR_ASSUMED_STATE] is True
+
+
+async def test_restore_unknown_state(hass: HomeAssistant):
+    """Tests restoring unknown entity state."""
+    disconnected_device = await create_mock_snooz(
+        connected=False, initial_state=UnknownSnoozState
+    )
+    restored_state = State("fan.test_restored", STATE_UNKNOWN)
+
+    with patch(
+        "homeassistant.helpers.restore_state.RestoreEntity.async_get_last_state",
+        return_value=restored_state,
+    ):
+        await create_mock_snooz_config_entry(hass, disconnected_device)
+
+        entity_id = get_fan_entity_id(hass, disconnected_device)
+
+        state = hass.states.get(entity_id)
+        assert state.state == STATE_UNKNOWN
+        assert state.attributes[ATTR_ASSUMED_STATE] is True
+
+
+async def test_command_results(
+    hass: HomeAssistant, mock_connected_snooz: SnoozFixture, snooz_fan_entity_id: str
+):
+    """Test device command results."""
+    with patch(
+        "homeassistant.components.snooz.fan.SnoozFan._async_write_state_changed",
```

Comment:
https://github.com/home-assistant/core/pull/78790/files#r990833713
https://developers.home-assistant.io/docs/development_testing?_highlight=testing#writing-tests-for-integrations
In this case you can check the result with `hass.states.get` instead